index/scorch/segment/zap/posting.go

```diff
@@ -617,6 +633,13 @@ func (i *PostingsIterator) nextDocNum() (uint64, bool, error) {
 	}
 
 	n := i.Actual.Next()
+	for uint64(n) < atOrAfter && i.Actual.HasNext() {
+		n = i.Actual.Next()
+	}
+	if uint64(n) < atOrAfter {
+		// couldn't find anything
+		return 0, false, nil
+	}
 	allN := i.all.Next()
```

Comment:
wondering the i.all iterator also needs the same treatment?

Reply:
hmmn, i think we have very different understandings of the way this works.  the purpose of the loop below is to safely allow "all" to catch up to where "n" is.  i say safely, because when we get to the right chunk we have to skip over freqs and locs, as dicated by the all iterator.  if we just jump it ahead here we're lost.

Reply:
ah, right